server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -702,29 +714,74 @@ private abstract static class IndexOptions implements ToXContent {
         }
     }
 
+    private record Int8ScalarQuantizationOptions(Float confidenceInterval) implements ToXContent {
+        static final String NAME = "byte";
+
+        static Int8ScalarQuantizationOptions fromParams(Map<String, ?> params) {
+            Object confidenceIntervalNode = params.remove("confidence_interval");
+            if (confidenceIntervalNode == null) {
+                return new Int8ScalarQuantizationOptions(null);
+            }
+            float confidenceInterval = (float) XContentMapValues.nodeDoubleValue(confidenceIntervalNode);
+            MappingParser.checkNoRemainingFields("quantization_options", params);
+            return new Int8ScalarQuantizationOptions(confidenceInterval);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            builder.field("type", NAME);
+            if (confidenceInterval != null) {
+                builder.field("confidence_interval", confidenceInterval);
+            }
+            builder.endObject();
+            return builder;
+        }
+    }
+
     private static class HnswIndexOptions extends IndexOptions {
         private final int m;
         private final int efConstruction;
+        private final Int8ScalarQuantizationOptions quantizationOptions;
 
         static IndexOptions parseIndexOptions(String fieldName, Map<String, ?> indexOptionsMap) {
             Object mNode = indexOptionsMap.remove("m");
             Object efConstructionNode = indexOptionsMap.remove("ef_construction");
+            Object quantizationOptionsNode = indexOptionsMap.remove("quantization_options");
             if (mNode == null) {
-                throw new MapperParsingException("[index_options] of type [hnsw] requires field [m] to be configured");
+                mNode = Lucene99HnswVectorsFormat.DEFAULT_MAX_CONN;
```

Comment:
We probably need a test that configures `m` and not `ef_construction` since it was not possible before this change?